src/terra-dev-site/guide/DockerDevelopmentEnvironment.i.guide.md

```diff
@@ -0,0 +1,91 @@
+## Docker Development Environment
+
+Terra provides a docker development environment that allows developers access to a common environment to develop Terra components regardless of the OS they are developing on.
+
+In each Terra repository there is a docker-compose.yml file that defines several services, most importantly ```dev``` and ```test```.
+
+To build all the services in the docker-compose file, run:
+```
+docker-compose build
+```
+
+### Dev
+The dev service provides a linux environment with node and npm pre-installed. By default the dev service volume mounts the entire code repo into the docker container to enable hot reloading during development.
+
+To get started developing, run:
+
+```bash
+docker-compose up dev
+```
+This will start the docker container for the repo, run npm install (if node_modules are not detected) and start the dev server. Then you're free to use your favorite editor to manipulate the code in the repo.
+
+### Test
+Terra-toolkit uses docker as part of how wdio tests are run. As a result as separate service is required to run wdio tests (jest or linting can be run via the terminal in the docker dev container.
+
+By default the test service in the terra repos will also start up a standalone chrome seleinum docker container to run wdio tests against.
+
+To test the repo, run:
+```bash
+docker-compose run test
+```
+This will start up the standalone chrome docker container, start the test service, run npm install and start running all tests in the repo. Similar to the dev service the entire code repo is volume mounted into the docker container. This is done in order to not require a rebuild of the docker container while developing tests.
```

Comment:
This is done ~in order to not require~ **to prevent** a rebuild of the docker

Reply:
Doesn't the run npm install only occur if local node-modules do not exist?

Reply:
Yes but that is mentioned below